files/image_config/caclmgrd/caclmgrd

```diff
@@ -358,64 +406,88 @@ class ControlPlaneAclManager(object):
         # This allows the device to respond to tools like tcptraceroute
-        iptables_cmds.append("iptables -A INPUT -m ttl --ttl-lt 2 -j ACCEPT")
-        iptables_cmds.append("ip6tables -A INPUT -p tcp -m hl --hl-lt 2 -j ACCEPT")
+        iptables_cmds.append(self.iptables_cmd_ns_prefix[namespace] + "iptables -A INPUT -m ttl --ttl-lt 2 -j ACCEPT")
+        iptables_cmds.append(self.iptables_cmd_ns_prefix[namespace] + "ip6tables -A INPUT -p tcp -m hl --hl-lt 2 -j ACCEPT")
 
         # Finally, if the device has control plane ACLs configured,
         # add iptables/ip6tables commands to drop all other incoming packets
         if num_ctrl_plane_acl_rules > 0:
-            iptables_cmds.append("iptables -A INPUT -j DROP")
-            iptables_cmds.append("ip6tables -A INPUT -j DROP")
+            iptables_cmds.append(self.iptables_cmd_ns_prefix[namespace] + "iptables -A INPUT -j DROP")
+            iptables_cmds.append(self.iptables_cmd_ns_prefix[namespace] + "ip6tables -A INPUT -j DROP")
 
         return iptables_cmds
 
-    def update_control_plane_acls(self):
+    def update_control_plane_acls(self, namespace):
         """
         Convenience wrapper which retrieves current ACL tables and rules from
         Config DB, translates control plane ACLs into a list of iptables
         commands and runs them.
         """
-        iptables_cmds = self.get_acl_rules_and_translate_to_iptables_commands()
-
+        iptables_cmds = self.get_acl_rules_and_translate_to_iptables_commands(namespace)
         log_info("Issuing the following iptables commands:")
         for cmd in iptables_cmds:
             log_info("  " + cmd)
 
         self.run_commands(iptables_cmds)
 
-    def notification_handler(self, key, data):
-        log_info("ACL configuration changed. Updating iptables rules for control plane ACLs...")
-        self.update_control_plane_acls()
-
     def run(self):
-        # Unconditionally update control plane ACLs once at start
-        self.update_control_plane_acls()
-
-        # Subscribe to notifications when ACL tables or rules change
-        self.config_db.subscribe(self.ACL_TABLE,
-                lambda table, key, data: self.notification_handler(key, data))
-        self.config_db.subscribe(self.ACL_RULE,
-                lambda table, key, data: self.notification_handler(key, data))
-
-        # Indefinitely listen for Config DB notifications
-        self.config_db.listen()
-
+        # Select Time-out for 10 Seconds
+        SELECT_TIMEOUT_MS = 1000 * 10
+
+        # Initlaize Global config that loads all database*.json
+        if device_info.is_multi_npu():
+            swsscommon.SonicDBConfig.initializeGlobalConfig()
+        # Create the Select object
+        sel = swsscommon.Select()
+        # Map of Namespace <--> susbcriber table's object
+        config_db_subscriber_table_map = {}
+        # Loop through all asic namespaces (if present) and host (namespace='')
```

Comment:
Please add a newline above this comment to visually break things up

Reply:
@jleveque Done